code/game/objects/items/teleportation.dm

```diff
@@ -234,3 +235,150 @@
 		else
 			itemUser.visible_message("<span class='suicide'>[user] looks even further depressed as they realize they do not have a head...and suddenly dies of shame!</span>")
 		return (BRUTELOSS)
+		
+/*
+ * Syndicate Teleporter
+ */
+
+/obj/item/teleporter
+	name = "syndicate teleporter"
+	desc = "A Syndicate reverse-engineered version of the Nanotrasen portable handheld teleporter. It uses bluespace technology to translocate users, but lacks the advanced safety features of its counterpart. Warranty voided if exposed to an electromagnetic pulse."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "syndi_tele"
+	item_state = "electronic"
+	lefthand_file = 'icons/mob/inhands/misc/devices_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/misc/devices_righthand.dmi'
+	throwforce = 5
+	w_class = WEIGHT_CLASS_SMALL
+	throw_speed = 4
+	throw_range = 10
+	item_state = "electronic"
+
+	var/charges = 4
+	var/max_charges = 4
+	var/minimum_teleport_distance = 4
+	var/maximum_teleport_distance = 8
+	var/saving_throw_distance = 3
+
+/obj/item/teleporter/Destroy()
+	STOP_PROCESSING(SSobj, src)
+	return ..()
+
+/obj/item/teleporter/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>[src] has [charges] out of [max_charges] charges left.</span>"
+
+/obj/item/teleporter/attack_self(mob/user)
+	..()
+	attempt_teleport(user, FALSE)
+
+/obj/item/teleporter/process()
+	if(charges < max_charges)
+		if(prob(10))
+			charges++
+	else
+		return PROCESS_KILL
```

Comment:
10% chance to charge every process seems pretty dodgy, would be much better if you instead had a set time between recharges and just used a timer to set the charge, removing the need for processing